Update customize_identity_model.md

Rick-Anderson opened the pull request:
[Internal review url](https://review.docs.microsoft.com/en-us/aspnet/identity/overview/features-api/customize_identity_model?branch=pr-en-us-7301)

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
```diff
@@ -1,13 +1,19 @@
 ---
 # Identity model customization
 
+https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.entityframeworkcore.identitydbcontext?view=aspnetcore-2.1
```
What is this link for?

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
```diff
@@ -16,19 +22,25 @@ ASP.NET Core Identity provides a framework for managing and storing user account
+ASP.NET Core has a development-time error page handler that can be used to apply migrations when the application is run. For production applications, it is often more appropriate to generate SQL scripts from the migrations and deploy these to the database as part of a controlled application and database deployment.
 
-The series of steps defined above will need to be repeated as additional changes are made to the model as described in this article.
+When a new application using Identity is created, steps 1 and 2 above have already been completed. That is, the initial data model already exists, and the initial migration has been added to the project. The initial migration still needs to be applied to the database.
```
Please add back in the commands to actually do this, otherwise people will struggle to find them.

Rick-Anderson replied:
@ajcvickers  do you want me to add back 
which can be done by running the Update-Database (PMC) or dotnet ef database update (.NET Core CLI) command, or by using the error page when the application is run.
?

ajcvickers replied:
Yes, please.

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
```diff
@@ -267,17 +279,17 @@ public abstract class IdentityUserContext<TUser, TKey, TUserClaim, TUserLogin, T
+* By overriding the OnModelCreating method to modify the mapping of these types.
 
-When overriding OnModelCreating, `base.OnModelCreating` should be called first, and then overriding configuration should be added next. This works because EF Core generally has a last-one-wins policy for configuration. For example, if the `ToTable` for an entity type is called first with one table name and then again later with a different table name, then the table name in the second call is the ...
+When overriding OnModelCreating, `base.OnModelCreating` should be called first, the overriding configuration should be called last. EF Core generally has a last-one-wins policy for configuration. For example, if the `ToTable` for an entity type is called first with one table name and then again later with a different table name, then the table name in the second call is the one that will be used.
```
Why last?

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
````diff
@@ -298,27 +310,27 @@ public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
 ```
 
-There is no need to override OnModelCreating here because EF Core will map the `CustomTag` property by convention. However, the database will need to be updated to get a new CustomTag column. To do this, add a migration, and then update the database as described at the top of this article.
+There is no need to override `OnModelCreating` here because EF Core will map the `CustomTag` property by convention. However, the database will need to be updated to use the new `CustomTag` column. For more information, see [Identity and EF Core Migrations](#identity-migrations).
````
Don't really understand this edit. What does it mean for a "database to use a column"?

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
```diff
@@ -775,9 +787,12 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
-Finally, let's add support for lazy-loading proxies in the Identity model. This can be very useful since it allows transparent navigation from on Identity entity to another related entity without first having to check whether the related entity has actually been loaded.
+In this section support for lazy-loading proxies in the Identity model is added. Lazy-loading proxies can be useful since it allows:
+
+* Transparent navigation from one Identity entity to another related entity.
```
This makes it sound like it is two ways in which lazy-loading can be useful. It's just one: that navigation can be done without first ensuring the navigations have been loaded.

Rick-Anderson replied:
Anything more than 25 words is difficult for native speakers and impossible for MT.

ajcvickers replied:
How about: "Lazy-loading is useful since it allows navigation properties to be used without first ensuring they are loaded." Captures what I wanted to say and only 18 words! ;-)

ajcvickers commented on aspnet/identity/overview/features-api/customize_identity_model.md:
```diff
@@ -775,9 +787,12 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
-Entity types can be made suitable for lazy-loading in several ways, as described in the [EF Core documentation](https://docs.microsoft.com/en-us/ef/core/querying/related-data#lazy-loading). For simplicity, we will use lazy-loading proxies, which requires:
+Entity types can be made suitable for lazy-loading in several ways, as described in the [EF Core documentation](/ef/core/querying/related-data#lazy-loading). For simplicity, we will use lazy-loading proxies, which requires:
 
 * Installation of the [Microsoft.EntityFrameworkCore.Proxies](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Proxies/)
```
Noticed typo here - missing the word "package".

Rick-Anderson commented:
@ajcvickers  if you can supply an app with 90% of the code in the article I can update the article to import snippets rather than copy/paste code. That makes it much easier to maintain.

ajcvickers commented:
@Rick-Anderson I can supply about 10 apps with all the variations, if that helps.

Rick-Anderson commented:
@ajcvickers  if you want to maintain 10 apps.  You don't need 100% import, so if it's easy to get 90% imported in less apps that's an option.

Rick-Anderson commented:
@ajcvickers  I think I've addressed all your comments. If so you can squash/merge this into your PR.
I'll do the snippet import at a later time in another PR.

ajcvickers commented:
@Rick-Anderson Okay, I'll see what I can do on the apps.

Rick-Anderson commented:
@ajcvickers  can I merge this into your #7288 PR?

ajcvickers commented:
@Rick-Anderson  Yes, go for it. Sorry, in meetings...
